### ISSUE TITLE
Decide the story about record fields.

### ISSUE THREAD

**Issue by robinp** [google/haskell-indexer#19]
See comments in indexer backend, Kythe verifier comments [1], also https://groups.google.com/forum/#!topic/kythe-dev/Mvus07b8c-U.

[1]: in kythe-verification/testdata/basic/RecordReadRef.hs, also RecordWriteRef.hs

**Comment by robinp** [comment]
Note: GHC 8 changes record field things (due to overloaded fields). Unless it would greatly simplify/unblock the GHC8 support, this change is better done after the GHC8 issue #2 is fixed, to avoid merge clash.

**Comment by robinp** [comment]
TLDR from above mail thread link:

- The "right" way to handle punned deconstructing ( `afun Foo{foo} = foo + 1`) would be that the first `foo` span both rerences the field and binds a local with name `foo`, and the second span would reference the local.

- Kythe prefers that the UI supports such multiple-things-at-span occurrences.

- Dilemma note 1: An usability dilemma is, that without punning (and assuming a multiple-line body with multiple `foo` refs) the individual `foo rec` lines would show up in backreferences, while with punning only the deconstructing line would show up. The gut feeling is that actual `foo rec` usages are more interesting to see in a backref listing.

- Dilemma note 2: the UI / serving backend could do the extra step of connecting the deconstructing binding with the refs and presenting them in the backref. However this likely means many additional calls to Kythe serving.


In the mean time there's a Typescript indexer, and Typescript has features like deconstructing (destructuring - https://basarat.gitbooks.io/typescript/docs/destructuring.html). Should have a look what the indexer emits in those cases.
